controller/main.go

```diff
@@ -74,22 +82,38 @@ func (c *controller) SetBalancer(l log.Logger, name string, svcRo *v1.Service, _
 	// a reason.
 	svc := svcRo.DeepCopy()
 	syncStateRes := controllers.SyncStateSuccess
-	wasAllocated := c.isServiceAllocated(name)
+
+	prevPool := c.ips.Pool(name)
+
 	if c.convergeBalancer(l, name, svc) != nil {
 		syncStateRes = controllers.SyncStateErrorNoRetry
 	}
 
-	if wasAllocated && !c.isServiceAllocated(name) { // convergeBalancer may deallocate our service and this means it did it.
-		// if the service was deallocated, it may have have left room
-		// for another one, so we reprocess
-		level.Info(l).Log("event", "serviceUpdated", "msg", "removed loadbalancer from service, services will be reprocessed")
-		syncStateRes = controllers.SyncStateReprocessAll
-	}
 	if reflect.DeepEqual(svcRo, svc) {
 		level.Debug(l).Log("event", "noChange", "msg", "service converged, no change")
 		return syncStateRes
 	}
 
+	newPool := c.ips.Pool(name)
+
+	// Generate pool status events.
+	if prevPool != newPool {
+		if prevPool != "" {
+			c.generatePoolStatusEvent(prevPool)
+		}
+		if newPool != "" {
+			c.generatePoolStatusEvent(newPool)
+		}
+	}
+
+	if prevPool != "" && newPool == "" {
```

Comment:
what do we gain with that? I think this is simple enough to understand

Reply:
It is simple enough but requires an effort that "wasAllocated" and isNotAllocated does not, and I don't think it's worth the performance gain.
This must be a general consideration for the whole project. No performance gain (that we know to be super significant) is worth a penalty in the readability and a few other complex ifs, to the point that I am tempted to say, let's always throw an event and iterate always all the pools status to check if something changed.

I'd rather consume a few more unnecessary events but having a code that is more explicit, is clearer to understand and that has less corner cases rather than super performant.

Reply:
alright, I'm adding it back